### ISSUE TITLE
From version3.x to version 6.x CentralProcesor getProcessorID is different from centralProcessor.getProcessorIdentifier().getProcessorID();

### ISSUE THREAD

**Issue by yzhang921** [oshi/oshi#2130]
From version3.x to version 6.x CentralProcesor getProcessorID is different from centralProcessor.getProcessorIdentifier().getProcessorID(); 

version 3.x: 
```
ProcessorID: 0F81FBFF00500657
manufacturer: OpenStack Foundation
model: OpenStack Nova (version: 13.2.1-20211018165854_809e3c6)
sno: 8e168e10-ca25-4403-a616-7622bbf3dbc5
firmware:
  manufacturer: SeaBIOS
  name: unknown
  description: dmi:bvnSeaBIOS:bvrrel-1.10.2-0-g5f4c7b1-20181220_000000-szxrtosci10000:bd04/01/2014:svnOpenStackFoundation:pnOpenStackNova:pvr13.2.1-20211018165854_809e3c6:cvn:ct1:cvrpc-i440fx-2.8:
  version: rel-1.10.2-0-g5f4c7b1-20181220_000000-szxrtosci10000
  release date: 2014-04-01
```

version 6.x
```
ProcessorID: 57 06 05 00 FF FB 8B 0F
manufacturer: OpenStack Foundation
model: OpenStack Nova (version: 13.2.1-20211018165854_809e3c6)
sno: 8e168e10-ca25-4403-a616-7622bbf3dbc5
firmware:
  manufacturer: unknown
  name: SMBIOS 2.8
  description: dmi:bvnSeaBIOS:bvrrel-1.10.2-0-g5f4c7b1-20181220_000000-szxrtosci10000:bd04/01/2014:svnOpenStackFoundation:pnOpenStackNova:pvr13.2.1-20211018165854_809e3c6:cvn:ct1:cvrpc-i440fx-2.8:
  version: rel-1.10.2-0-g5f4c7b1-20181220_000000-szxrtosci10000
  release date: 2014-04-01
```

**Comment by dbwiddis** [comment]
TLDR: Your "3.x" code is from 3.5.0 or earlier and had a bug fixed in 2018.

[Here's the 6.X code](https://github.com/oshi/oshi/blob/a6aa03f9a181e7f9e02efcbe26b17a043605d7c3/oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxCentralProcessor.java#L446-L478).  OSHI:
1. Tries `dmidecode -t 4` (which requires root/sudo) and collects the string following "ID:".  This is the spaced output you see in your 6.x output, indicating that you were running with elevated permissions that dmidecode requires.
2. Tries `cpuid -1r` (which requires that you've installed the `cpuid` package) which returns the actual byte coded result of the CPUID instruction.  I think this might be what you're seeing in your 3.x output.  I'm not sure why there's a one bit difference in that, though.  Maybe it's not.
3. Calculates a best guess at the CPUID from the family, stepping, model, and flags information available.  This might also be the output you are seeing in your 3.x version since I know there is one bit that can't be determined from that info, so it guesses, and that's why in the 3.x output you see 81 instead of 8B in the `dmidecode` output.

`git blame` shows that section of code hasn't changed since #480 which fixed the bug reported in #430 where the `dmidecode` portion wasn't being fetched properly.  While that didn't seem to make the change log, it was apparently fixed between versions 3.5.0 and 3.6.0.

So the "3.x" output you are seeing is the fallback guess which is inaccurate; output after 3.6.0 (with elevated permissions) should give you the correct result.

The difference in format is noted in the [Javadoc for getProcessorId()](https://www.oshi.ooo/oshi-core-java11/apidocs/com.github.oshi/oshi/hardware/CentralProcessor.ProcessorIdentifier.html#getProcessorID()):
> NOTE: The order of returned bytes is platform and software dependent. Values may be in either Big Endian or Little Endian order.
> 
> NOTE: If OSHI is unable to determine the ProcessorID from native sources, it will attempt to reconstruct one from available information in the processor identifier.

See also a [related discussion here](https://github.com/oshi/oshi/discussions/1863#discussioncomment-2025341).

**Comment by yzhang921** [comment]
@dbwiddis thanks for your detailed explanation, looks like I have to abandon processId as one Identifier. 
BTW, does `systemInfo.getHardware.getComputerSystem.getSerialNumber` can use as hardware ID, sometimes it would retruen 'unknown'

**Comment by dbwiddis** [comment]
> @dbwiddis thanks for your detailed explanation, looks like I have to abandon processId as one Identifier. 

Yeah, particularly as non-Intel chips become more prevelant, this really isn't a useful tool.   One of many possibilities but certainly not a consistent one, particularly if someone can change the result by just installing software.  

> BTW, does `systemInfo.getHardware.getComputerSystem.getSerialNumber` can use as hardware ID, sometimes it would retruen 'unknown'

This is another case where OSHI does its best to give a value but Linux has gone to great lengths for privacy and makes it really difficult to get unique identifiers without root/elevated permissions.   When you get it, you can rely on it.  When you don't... you'll get "unknown" which is at least consistent but not useful for licensing applications.
